service/header/store.go

```diff
@@ -120,6 +110,11 @@ func (s *store) Get(_ context.Context, hash tmbytes.HexBytes) (*ExtendedHeader,
 }
 
 func (s *store) GetByHeight(ctx context.Context, height uint64) (*ExtendedHeader, error) {
+	h, err := s.heightSub.Sub(ctx, height)
+	if h != nil || err != nil {
```

Comment:
What is `h` and why do we return if it is != nil? 

Ah, I see the extended header. The flow here is not obvious: So we either return if we get and extended header on subscribe (or error), or, we have get it from the store (below)? I think comments would help here.

Reply:
Commented.

Reply:
Good comment. I'd usually prefer if stuff like this was expressed via code directly. But I think this is good as is.

Reply:
Changed to error